### PR TITLE
Ported to android-21

### PR DESCRIPTION
**Pull request by alterakey**
Firstly, thank you for putting great Material example code in public!
I tried to build your project against android-21.
Would you please review my changes, and hopefully merge this..?

Thanks in advance.

### CODE REVIEW

File: app/src/main/java/com/example/android/io2014/DetailActivity.java [hunk 1]
```diff
@@ -116,13 +117,17 @@ private void setupMap() {
...
 
     private void setOutlines(int star, int info) {
-        int size = getResources().getDimensionPixelSize(R.dimen.floating_button_size);
+        final int size = getResources().getDimensionPixelSize(R.dimen.floating_button_size);
 
-        Outline outline = new Outline();
-        outline.setOval(0, 0, size, size);
+        final ViewOutlineProvider vop = new ViewOutlineProvider() {
+            @Override
+            public void getOutline(final View view, final Outline outline) {
```
Comment (romainguy) [inline app/src/main/java/com/example/android/io2014/DetailActivity.java]: Don't make the parameters final

File: app/src/main/java/com/example/android/io2014/ui/CardFrameLayout.java [hunk 2]
```diff
@@ -34,15 +36,21 @@ public CardFrameLayout(Context context, AttributeSet attrs, int defStyle) {
     }
 
     @Override
-    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
+    protected void onSizeChanged(final int w, final int h, final int oldw, final int oldh) {
```
Comment (romainguy) [inline app/src/main/java/com/example/android/io2014/ui/CardFrameLayout.java]: Don't make the parameters final

File: app/src/main/java/com/example/android/io2014/ui/CardFrameLayout.java [hunk 3]
```diff
@@ -34,15 +36,21 @@ public CardFrameLayout(Context context, AttributeSet attrs, int defStyle) {
...
         super.onSizeChanged(w, h, oldw, oldh);
 
         final float radius = getResources().getDimensionPixelSize(R.dimen.card_corner_radius);
 
-        Outline outline = new Outline();
-        outline.setRoundRect(0, 0, w, h, radius);
+        final ViewOutlineProvider vop = new ViewOutlineProvider() {
+            @Override
+            public void getOutline(final View view, final Outline outline) {
```
Comment (romainguy) [inline app/src/main/java/com/example/android/io2014/ui/CardFrameLayout.java]: Don't make the parameters final

**Comment by alterakey** [comment]
Thanks for the review!
Applied your comments.

**Comment by romainguy** [comment]
Thank you so much for this pull request!

**Pull request merged**